portafly/src/tests/components/shared/data-list/DataListContext.test.tsx

```diff
@@ -54,3 +56,59 @@ describe('DataListContext: useDataFilters', () => {
     expect(filters.innerHTML).toBe('')
   })
 })
+
+describe('DataListContext: DataReducer', () => {
+  const ExampleComponent = () => {
+    const { data, setData } = useDataListData()
+    const [adminName, setAdminName] = useState('')
+    const onChange = (ev: React.ChangeEvent<HTMLInputElement>) => (
+      setAdminName(ev.currentTarget.value)
+    )
+
+    const addAccount = () => (
+      setData(
+        [...data, ...[factories.DeveloperAccount.build({ adminName })]] as IDeveloperAccount[]
```

Comment:
I think this would not be needed if we use generics
```suggestion
        [...data, ...[factories.DeveloperAccount.build({ adminName })]]
```